### ISSUE TITLE
[a11y]: Add more explicit roles or labels to tearsheet regions

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#7850]
### Package

@carbon/ibm-products

### Package version

v2.70.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Visually the tearsheet has 2 sections, which are
1. influencer information in the left pane,
2. Main content in the right pane

However, the screen reader is not able to differentiate between these 2 content panels, and unable to identify the boundaries on arrow key navigation.

**Recommended fix**

For the benefits of screen reader users, recommend that we define these 2 content sections into 2 different labelled regions, for example,
1. Left pane (influencer section) as “influencer region,” and 
2. Right pane (main content section) as “main region”
 
Use role region with suitable labels.

**User impact**
When the page content is not covered within a labelled landmark region, screen reader users are unable to skim through the page and understand its structure.

> [!NOTE]
> Our Storybook has since added headings in these examples, but we’ve also seen examples in our own docs and websites using navigational elements or progress indicators in lieu of structured headings.

### WCAG 2.1 Violation

2.4.1 Bypass Blocks

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/components-tearsheet--with-influencer

### Steps to reproduce

To reproduce this defect:

1. Use a screen reader and navigate between the influencer and main content regions.
2. While the headings and content do get announced, no regions are specified.

This can be more challenging for screen reader users to understand the structure of page content when headings are not included.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

### Acceptance criteria

- [ ] Evaluate whether to add explicit roles to the influencer/content areas or add support for aria-labels (investigate aside within dialog. roles within roles? nested roles??)
- [ ] Update the influencer and content regions based on evaluation

These will also be considerations for the composable tearsheet.
